dali/kernels/transpose/transpose_gpu_impl.cuh

```diff
@@ -148,10 +148,15 @@ __device__ void TransposeTiledStatic(TiledTransposeDesc<T> desc) {
     in_ofs  += desc.in_strides[ndim-2]  * in_y  + desc.in_strides[ndim-1]  * in_x;
     out_ofs += desc.out_strides[ndim-2] * out_y + desc.out_strides[ndim-1] * out_x;
 
+    in_ofs >>= pack_ratio_log;
+    out_ofs >>= pack_ratio_log;
+
     unsigned tile_w = min(static_cast<uint64_t>(kTileSize), desc.shape[ndim-1] - pos[ndim-1]);
     unsigned tile_h = min(static_cast<uint64_t>(kTileSize), desc.shape[ndim-2] - pos[ndim-2]);
+    unsigned jump = blockDim.y >> pack_ratio_log;
```

Comment:
Can we be sure that it is at least 4?

Reply:
In fact, would keeping `dy` jumps as they were (and scaling here the `desc.in_strides[ndim-2]` instead) make a (performance) difference? Personally, I was a bit surprised at first seeing that the `y` jump changes with this optimization.

Reply:
I checked it now and i think we can't be sure, but from my understanding in this scenario old code wouldn't work as intended (the same values would be copied more than one time). I don't know if MaxThreadsPerBlock function can in real scenario return something less than 64 (or rather 32*numeber_of_lanes) , if no then the code should be safe, if yes then I added the critical bug, and there was performance bug before

Reply:
About the performance I can change it to scaling the strides, it does not change the performance, only change is that i have to add two variables instead of one (doing it without variables loses 25us in benchmark time). 
There is also a third way, I can scale strides in the beginning(editing the desc), this could add minimal increase in performance, but I don't know if editing desc is a good idea

Reply:
changed for recalculating strides

Reply:
> Can we be sure that it is at least 4 (or to be exact: `2 ** pack_ratio_log`)?

about that, if I calculate correctly this blockDim.y is bound by 1024/used_registers, where used registers in this case is less than 200 so for me code is safe